configapi/api_inventory.go

```diff
@@ -0,0 +1,231 @@
+    "go.mongodb.org/mongo-driver/bson"
+)
+
+const (
+    gnbDataColl = "webconsoleData.snapshots.gnbData"
+    upfDataColl = "webconsoleData.snapshots.upfData"
+)
+
+func setInventoryCorsHeader(c *gin.Context) {
+    c.Writer.Header().Set("Access-Control-Allow-Origin", "*")
+    c.Writer.Header().Set("Access-Control-Allow-Credentials", "true")
+    c.Writer.Header().Set("Access-Control-Allow-Headers", "Content-Type")
+    c.Writer.Header().Set("Access-Control-Allow-Methods", "POST, GET, DELETE")
+}
+
+func GetGnbs(c *gin.Context) {
+    setInventoryCorsHeader(c)
+    logger.WebUILog.Infoln("Get all gNBs")
+
+    var gnbs []*configmodels.Gnb
+    gnbs = make([]*configmodels.Gnb, 0)
+    rawGnbs, errGetMany := dbadapter.CommonDBClient.RestfulAPIGetMany(gnbDataColl, bson.M{})
+    if errGetMany != nil {
+        logger.DbLog.Errorln(errGetMany)
+        c.JSON(http.StatusInternalServerError, gnbs)
+    }
+
+    for _, rawGnb := range rawGnbs {
+        var gnbData configmodels.Gnb
+        err := json.Unmarshal(mapToByte(rawGnb), &gnbData)
+        if err != nil {
+            logger.DbLog.Errorf("Could not unmarshall gNB %v", rawGnb)
+        }
+        gnbs = append(gnbs, &gnbData)
+    }
+    c.JSON(http.StatusOK, gnbs)
+}
+
+func GnbPostByName(c *gin.Context) {
+    setInventoryCorsHeader(c)
+    if err := gnbPostHandler(c); err == nil {
+        c.JSON(http.StatusOK, gin.H{})
+    } else {
+        c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+    }
+}
+
+func GnbDeleteByName(c *gin.Context) {
+    setInventoryCorsHeader(c)
+    if err := gnbDeletetHandler(c); err == nil {
+        c.JSON(http.StatusOK, gin.H{})
+    } else {
+        c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+    }
+}
+
+func gnbPostHandler(c *gin.Context) error {
+    var gnbName string
+    var exists bool
+    if gnbName, exists = c.Params.Get("gnb-name"); !exists {
+        errorMessage := "Post gNB request is missing gnb-name"
+        configLog.Errorf(errorMessage)
+        return errors.New(errorMessage)
+    }
+    configLog.Infof("Received gNB %v", gnbName)
+    var err error
+    var newGnb configmodels.Gnb
+
+    allowHeader := strings.Split(c.GetHeader("Content-Type"), ";")
+    switch allowHeader[0] {
+    case "application/json":
+        err = c.ShouldBindJSON(&newGnb)
+    }
+    if err != nil {
+        configLog.Errorf("err %v", err)
+        return fmt.Errorf("Failed to create gNB %v: %w", gnbName, err)
+    }
+    if newGnb.Tac == "" {
+        errorMessage := "Post gNB request body is missing tac"
+        configLog.Errorf(errorMessage)
+        return errors.New(errorMessage)
+    }
+    req := httpwrapper.NewRequest(c.Request, newGnb)
+    procReq := req.Body.(configmodels.Gnb)
+    procReq.GnbName = gnbName
+    msg := configmodels.ConfigMessage{
+        MsgType:     configmodels.Inventory,
+        MsgMethod:   configmodels.Post_op,
+        GnbName: gnbName,
+        Gnb: &procReq,
+    }
+    configChannel <- &msg
```

Comment:
This is probably a lack of understanding on my part but why do we use this `configChannel`?

Reply:
Let's keep all config flowing through configChannel so all handling is done at same place.